src/server/admin/server/admin_test.go

```diff
@@ -192,22 +193,52 @@ func RepoInfoToName(repoInfo interface{}) interface{} {
 	return repoInfo.(*pfs.RepoInfo).Repo.Name
 }
 
-// testExtractRestore effectively implements both TestExtractRestoreObjects
-// TestExtractRestoreNoObjects, as their logic is mostly the same
-func testExtractRestore(t *testing.T, testObjects bool) {
-	if os.Getenv("RUN_BAD_TESTS") == "" {
-		t.Skip("Skipping because RUN_BAD_TESTS was empty")
-	}
+// testExtractRestore effectively implements TestExtractRestoreObjects,
+// TestExtractRestoreNoObjects, TestExtractRestoreAuth, as their logic is mostly the same
+func testExtractRestore(t *testing.T, testObjects, testAuth bool) {
 	if testing.Short() {
 		t.Skip("Skipping integration tests in short mode")
 	}
 
 	c := tu.GetPachClient(t)
 	require.NoError(t, c.DeleteAll())
 
+	// If we're testing auth, get an authenticated client.
+	// This hash the side-effect of enabling enterprise and auth
+	if testAuth {
+		c = tu.GetAuthenticatedPachClient(t, tu.AdminUser)
+		defer tu.DeleteAll(t)
+	}
+
 	dataRepo := tu.UniqueString("TestExtractRestoreObjects-in-")
 	require.NoError(t, c.CreateRepo(dataRepo))
 
+	// If we're testing auth, give all the pipeline users access to the first repo
+	// Also make some admin and FS admin users
+	if testAuth {
+		_, err := c.SetACL(c.Ctx(), &auth.SetACLRequest{
+			Repo: dataRepo,
+			Entries: []*auth.ACLEntry{
+				&auth.ACLEntry{Username: "github:pipeline-user-0", Scope: auth.Scope_READER},
+				&auth.ACLEntry{Username: "github:pipeline-user-1", Scope: auth.Scope_WRITER},
+				&auth.ACLEntry{Username: "github:pipeline-user-2", Scope: auth.Scope_OWNER},
```

Comment:
All these users don't _need_ access to the initial repo, but we can assert that this ACL is restored correctly later.